pkg/artifactory/resource/repository/federated/federated.go

```diff
@@ -113,3 +129,60 @@ func PackMembers(members []Member, d *schema.ResourceData) error {
 
 	return nil
 }
+func deleteRepo(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// For federated repositories we delete all the federated members (except the initial repo member), if the flag `cleanup_on_delete` is set to `true`
+	s := &util.ResourceData{ResourceData: d}
+	initialRepoName := s.GetString("key", false)
+	if v, ok := d.GetOk("member"); ok && s.GetBool("cleanup_on_delete", false) {
+		// Save base URL from the Client to be able to revert it back after the change below
+		baseURL := m.(util.ProvderMetadata).Client.BaseURL
+		federatedMembers := v.(*schema.Set).List()
+		for _, federatedMember := range federatedMembers {
+			id := federatedMember.(map[string]interface{})
+			memberUrl := id["url"].(string) // example "https://artifactory-instance.com/artifactory/federated-generic-repository-example"
```

Comment:
Next time when we touch these code, we should rename this to follow Golang convention, i.e. `memberURL` 😄